src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs

```diff
@@ -254,35 +255,29 @@ protected void UpdateSpans_CallOnlyOnUIThread(NormalizedSnapshotSpanCollection c
             AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(changedSpanCollection);
         }
 
-        protected bool ShouldDrawTag(SnapshotSpan snapshotSpan, IMappingTagSpan<T> mappingTagSpan, out IWpfTextViewLine viewLine)
```

Comment:
broke this helper into separate helpers for separate concerns.  Different subclasses use different pieces.